Tests/SonarScanner.MSBuild.Tasks.UnitTests/GetAnalyzerSettingsTests.cs

```diff
@@ -305,144 +273,140 @@ public void GetAnalyzerSettings_ConfigExists_NewBehaviour_SettingsMerged()
-                                new string[] { "c:\\wintellect1.dll", "c:\\wintellect\\bar.ps1", "c:\\Google.Protobuf.dll" }),
-
-                            new AnalyzerPlugin("vbnet", "1.1", "dummy resource2",
-                                new string[] { "c:\\sonar.vbnet1.dll", "c:\\foo.ps1", "c:\\Google.Protobuf.dll" }),
-
+                            new AnalyzerPlugin("roslyn.wintellect", "2.0", "dummy resource", new [] { @"c:\wintellect1.dll", @"c:\wintellect\bar.ps1", @"c:\Google.Protobuf.dll" }),
+                            new AnalyzerPlugin("vbnet", "1.1", "dummy resource2", new [] { @"c:\sonar.vbnet.dll", @"c:\foo.ps1", @"c:\Google.Protobuf.dll" }),
                         },
-
-                        AdditionalFilePaths = new List<string> { "c:\\add1.txt.vb", "d:\\replaced1.txt" }
+                        AdditionalFilePaths = new List<string> { @"c:\add1.vbnet.txt", @"d:\replaced1.txt" }
                     },
-
                     new AnalyzerSettings // Settings for a different language
                     {
                         Language = "cobol",
-                        RulesetPath = "f:\\xxx.ruleset",
-                        DeactivatedRulesetPath = "c:\\cobol\\test.project.ruleset",
+                        RulesetPath = @"c:\cobol-normal.ruleset",
+                        DeactivatedRulesetPath = @"c:\cobol-deactivated.ruleset",
                         AnalyzerPlugins = new List<AnalyzerPlugin>
                         {
-                            new AnalyzerPlugin("cobol.analyzer", "1.0", "dummy resource",
-                                new string[] { "c:\\cobol1.dll", "c:\\cobol2.dll" })
+                            new AnalyzerPlugin("cobol.analyzer", "1.0", "dummy resource", new [] { @"c:\cobol1.dll", @"c:\cobol2.dll" })
                         },
-
-                        AdditionalFilePaths = new List<string> { "c:\\cobol.\\add1.txt", "d:\\cobol\\add2.txt" }
+                        AdditionalFilePaths = new List<string> { @"c:\cobol.\add1.txt", @"d:\cobol\add2.txt" }
                     }
                 }
             };
 
             var testSubject = CreateConfiguredTestSubject(config, language, TestContext);
-            testSubject.IsTestProject = true;
-            testSubject.OriginalAnalyzers = new string[]
+            testSubject.IsTestProject = isTestProject;
+            testSubject.OriginalAnalyzers = new[]
             {
                  "c:\\analyzer1.should.be.replaced.dll",
                  "c:\\analyzer2.should.be.replaced.dll",
                  "c:\\Google.Protobuf.dll", // same name as an assembly in the csharp plugin (above)
             };
-            testSubject.OriginalAdditionalFiles = new string[]
+            testSubject.OriginalAdditionalFiles = new[]
             {
-                "original.should.be.removed.txt",
+                isTestProject ? "original.should.be.removed.for.excluded.test.txt" : "original.should.be.preserved.for.product.txt",
```

Comment:
why is there a difference here between Test and Product?
(maybe it's worth commenting)

Reply:
The values `original.should.be.removed.for.excluded.test` and `original.should.be.preserved.for.product.txt` are explaining that by themselves 😃 The value could be `foo.txt` in both cases, this will help a little with understanding the UT failure if something breaks.